hipe4ml/tree_handler.py

```diff
@@ -0,0 +1,278 @@
+            preselected df
+
+        Returns
+        ------------------------------------------------
+        out: pandas.DataFrame or None
+            If inplace == True return None is returned and the full DataFrame is replaced
+
+        """
+        df_pres = None
+        if inplace:
+            self._preselections = preselections
+            self._full_data_frame = self._full_data_frame.query(
+                self._preselections)
+            return df_pres
+
+        df_pres = self._full_data_frame.query(preselections)
+        return df_pres
+
+    def slice_data_frame(self, projection_variable, projection_binning, delete_original_df=False):
+        """
+        Create a list containing slices of the orginal DataFrame.
+        The original DataFrame is splitted in N sub-DataFrames following
+        the binning(projection_binning) of a given variable(projected_variable)
+
+        Parameters
+        ------------------------------------------------
+        projection_variable: str
+            Name of the variable that will be sliced in the analysis
+        projection_binning: list
+            Binning of the sliced variable should be given as a list of
+            [min, max) values for each bin
+
+        delete_original_df: bool
+            If True delete the original DataFrame. Only the
+            the slice array will be accessible in this case
+
+        """
+
+        self._projection_variable = projection_variable
+        self._projection_binning = projection_binning
+
+        self._sliced_df_list = []
+        for ibin in projection_binning:
+            bin_mask = np.logical_and(
+                self._full_data_frame[projection_variable] >= ibin[0],
+                self._full_data_frame[projection_variable] < ibin[1])
+            self._sliced_df_list.append(self._full_data_frame[bin_mask])
+        if delete_original_df:
+            self._full_data_frame = None
+
+    def shuffle_data_frame(self, size=None, frac=None, inplace=True):
+        """
+        Extract a random sample from the DataFrame
+
+        Parameters
+        ------------------------------------------------
+        size: int
+            Number of candidates to return. Cannot be used with frac. Default = 1 if
+            frac = None.
+
+        frac: float
+            Fraction of candidates to return. Cannot be used with n.
+
+        inplace: bool
+            If True the shuffled dataframe replaces the initial dataframe. Otherwise return a copy
+            of the shuffled df
+
+        Returns
+        ------------------------------------------------
+        out: pandas.DataFrame or None
+            If inplace == True None is returned and the full DataFrame is replaced
+
+        """
+        df_shuf = None
+        if inplace:
+            self._full_data_frame = self._full_data_frame.sample(size, frac)
+            return df_shuf
+
+        df_shuf = self._full_data_frame.sample(size, frac)
```

Comment:
It could be useful to be able to change the seed of the sampling, maybe some variables/kwargs of sample can be exposed in the function.